Tasks/Common/webdeployment-common/deployusingmsdeploy.ts

```diff
@@ -46,21 +43,7 @@ export async function DeployUsingMSDeploy(webDeployPkg, webAppName, publishingPr
     }
     catch(error) {
         tl.error(tl.loc('Failedtodeploywebsite'));
-        isDeploymentSuccess = false;
-        deploymentError = error;
         msDeployUtility.redirectMSDeployErrorToConsole();
-    }
-
-    if(publishingProfile != null){
-        try {
-            tl._writeLine(await azureRESTUtility.updateDeploymentStatus(publishingProfile, isDeploymentSuccess));
-        }
-        catch(error) {
-            tl.warning(error);
-        }
-    }
-
-    if(!isDeploymentSuccess) {
-        throw Error(deploymentError);
+        throw Error(error);
```

Comment:
the IISWebAppDeploymentOnMachine group task does not use updateDeploymentStatus function as publishing profile is always sent as null